src/urllib3/util/ssl_.py

```diff
@@ -356,5 +367,15 @@ def is_ipaddress(hostname):
     if six.PY3 and isinstance(hostname, bytes):
         # IDN A-label bytes are ASCII compatible.
         hostname = hostname.decode('ascii')
-
     return _IP_ADDRESS_REGEX.match(hostname) is not None
+
+
+def _is_key_file_encrypted(key_file):
+    """Detects if a key file is encrypted or not."""
+    with open(key_file, 'r') as f:
```

Comment:
@sethmlarson There would not be a way to put this in a try...except or something?  We want to avoid files or temporary files and we put a special HTTPAdapter to pass the contents immediately instead of the files, but then it tracebacks at this point.  Ideally, there would be support for passing the data immediately instead of opening files.

Reply:
@jco-odoo Can you open a separate issue for this?

Reply:
https://github.com/urllib3/urllib3/issues/2508